src/textual/widgets/_month_calendar.py

```diff
@@ -0,0 +1,379 @@
+            )
+        return first_weekday
+
+    def watch_date(self, old_date: datetime.date, new_date: datetime.date) -> None:
+        if not self.is_mounted:
+            return
+        if (new_date.month != old_date.month) or (new_date.year != old_date.year):
+            self._calendar_dates = self._get_calendar_dates()
+            self._update_calendar_table(update_week_header=False)
+        else:
+            table = self.query_one(MonthCalendarTable)
+            cursor_row, cursor_column = table.cursor_coordinate
+            if self._calendar_dates[cursor_row][cursor_column] != new_date:
+                date_coordinate = self._get_date_coordinate(self.date)
+                table.cursor_coordinate = date_coordinate
+
+    def watch_first_weekday(self) -> None:
+        self._calendar = calendar.Calendar(self.first_weekday)
+        self._calendar_dates = self._get_calendar_dates()
+        if not self.is_mounted:
+            return
+        self._update_calendar_table(update_week_header=True)
+
+    def watch_show_cursor(self, show_cursor: bool) -> None:
+        if not self.is_mounted:
+            return
+        table = self.query_one(MonthCalendarTable)
+        table.show_cursor = show_cursor
+
+    def watch_show_other_months(self) -> None:
+        self._calendar_dates = self._get_calendar_dates()
+        if not self.is_mounted:
+            return
+        self._update_calendar_table(update_week_header=False)
+
+    def action_select_date(self) -> None:
+        table = self.query_one(MonthCalendarTable)
+        table.action_select_cursor()
+
+    def action_previous_week(self) -> None:
+        table = self.query_one(MonthCalendarTable)
+        if self.show_other_months and table.cursor_row == 0:
+            self.date -= relativedelta(weeks=1)
+        else:
+            table.action_cursor_up()
+
+    def action_next_week(self) -> None:
+        table = self.query_one(MonthCalendarTable)
+        if self.show_other_months and table.cursor_row == table.row_count - 1:
+            self.date += relativedelta(weeks=1)
+        else:
+            table.action_cursor_down()
+
+    def action_next_day(self) -> None:
+        table = self.query_one(MonthCalendarTable)
+        if table.cursor_column == len(table.columns) - 1:
+            next_date = self.date + relativedelta(days=1)
+            if self.show_other_months or next_date.month == self.date.month:
+                self.date = next_date
+        else:
+            table.action_cursor_right()
+
+    def action_previous_day(self) -> None:
+        table = self.query_one(MonthCalendarTable)
+        if table.cursor_column == 0:
+            previous_date = self.date - relativedelta(days=1)
+            if self.show_other_months or previous_date.month == self.date.month:
+                self.date = previous_date
+        else:
+            table.action_cursor_left()
+
+    def action_next_month(self) -> None:
+        if self.show_other_months:
```

Comment:
It feels like the `show_other_months` check should be inside `next_month` maybe?

I'm actually wondering if `show_other_months` should just be removed. Does restricting the view to a single month have a common enough use-case?